pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -93,63 +114,27 @@ func machinesEqual(m1 *clusterv1.Machine, m2 *clusterv1.Machine) bool {
 	return m1.Name == m2.Name && m1.Namespace == m2.Namespace
 }
 
-// isNodeJoin determines if a machine, in scope, should join of the cluster.
-// TODO: Make this thread safe kubernetes-sigs/cluster-api#925
-// https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/745#discussion_r280506890
-func (a *Actuator) isNodeJoin(scope *actuators.MachineScope, controlPlaneMachines []*clusterv1.Machine) (bool, error) {
-	switch set := scope.Machine.ObjectMeta.Labels["set"]; set {
-	case "node":
-		// Worker machines, not part of the controlplane, will always join the cluster.
-		return true, nil
-	case "controlplane":
-		// Controlplane machines will join the cluster if the cluster has an existing control plane.
-		controlplaneExists := false
-		var err error
-
-		var sharedScope *actuators.MachineScope
-
-		for _, cm := range controlPlaneMachines {
-			if sharedScope == nil {
-				sharedScope, err = actuators.NewMachineScope(actuators.MachineScopeParams{
-					Machine:    cm,
-					Cluster:    scope.Cluster,
-					Client:     a.clusterClient,
-					Logger:     a.log,
-					AWSClients: scope.AWSClients,
-				})
-				if err != nil {
-					return false, errors.Wrapf(err, "failed to create machine scope for %s/%s", cm.Namespace, cm.Name)
-				}
-			}
-			sharedScope.Machine = cm
-			ec2svc := ec2.NewService(sharedScope.Scope)
-
-			controlplaneExists, err = ec2svc.MachineExists(sharedScope)
-			if err != nil {
-				return false, errors.Wrapf(err, "failed to verify existence of machine %s/%s", sharedScope.Machine.Namespace, sharedScope.Machine.Name)
-			}
-
-			if controlplaneExists {
-				return controlplaneExists, err
-			}
-		}
-
-		a.log.V(2).Info("Will machine join the controlplane", "machine-name", scope.Machine.Name, "machine-namespace", scope.Machine.Name, "should-join-control-plane", controlplaneExists)
-		return controlplaneExists, err
-
-	default:
-		return false, errors.Errorf("Unknown value %q for label `set` on machine %q", set, scope.Machine.Name)
-	}
-}
+const (
+	waitForClusterInfrastructureReadyDuration   = 15 * time.Second
+	waitForControlPlaneMachineExistenceDuration = 5 * time.Second
+	waitForControlPlaneReadyDuration            = 5 * time.Second
```

Comment:
Same as above, move these at the top of the file?

Reply:
I like them close to where they're accessed, but I can move them if you feel strongly.

Reply:
Not a huge problem, was pointing out for consistency's sake